src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright © 2021 Oscar Gómez (ogomezso0@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.github.ogomezso.kafka.connect.soap.client;
+
+import javax.xml.namespace.QName;
+import javax.xml.transform.stream.StreamSource;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.time.LocalDateTime;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.github.jcustenborder.kafka.connect.utils.config.ConfigUtils;
+import com.github.ogomezso.kafka.connect.soap.source.SoapSourceConnectorConfig;
+
+import jakarta.xml.soap.MessageFactory;
+import jakarta.xml.soap.SOAPException;
+import jakarta.xml.soap.SOAPMessage;
+import jakarta.xml.soap.SOAPPart;
+import jakarta.xml.ws.Dispatch;
+import jakarta.xml.ws.Service;
+import jakarta.xml.ws.soap.SOAPBinding;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SoapClient {
+
+  private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(4);
```

Comment:
consider retrieving the pool size dynamically from the runtime or configuration. Also double check that you are following executor sizing best practices  for CPU vs IO bound workloads https://gist.github.com/djspiewak/46b543800958cf61af6efa8e072bfd5c

Reply:
DONE. 

Retrieving poolsize on runtime and assign half to the thread pool size